olive/data/component/load_dataset.py

```diff
@@ -32,7 +34,8 @@ def huggingface_dataset(data_dir, data_name=None, subset=None, split="validation
     from datasets import load_dataset
 
     assert data_name is not None, "Please specify the data name"
-    return load_dataset(path=data_name, name=subset, split=split, data_files=data_files, **kwargs)
+    data_files = str(Path(data_dir)) if data_dir and Path(data_dir).is_file() else data_files
```

Comment:
>data_files = str(Path(data_dir)) if data_dir and Path(data_dir).is_file() else data_files [](http://example.com/codeflow?start=4&length=89)

why do we need this kinds of logic? I think we can simply pass both the data_dir and data_files to Huggingface.

Reply:
Sometime the data_dir actually a datafile, for instance:
```
                "data_dir": {
                    "type": "azureml_datastore",
                    "config": {
                        "datastore_name": "<datastore_name>",
                        "relative_path": "a_json_file.json"
                    }
                },
```
In this case, the `data_dir` will be a json file path. While `load_dataset` func `data_dir` param cannot recognize a file path to load the data.

Reply:
In this case, could we just pass the data directory downloaded from AzureML to Olive? the data_dir could be either files or dir is confusing to end user.

Reply:
What do you mean by this "pass the data directory downloaded from AzureML to Olive"?

Reply:
from the name, data_dir represent data directory. In azureml case, it should be the data directory contains the files.

Reply:
So how can we handle the case that the data on AML is a datafile?

Reply:
1. As Mike mentioned, we should not download the aml dataset and upload it again if the job run in aml.
2. If the data is on aml, but we need run olive locally, maybe user can download the data by themselves?
3. To avoid the misconception on data dir and files, can we directly leverage the `data_files` argument?